inc/sitemaps/class-author-sitemap-provider.php

```diff
@@ -131,8 +130,7 @@ protected function get_users( $arguments = [] ) {
 
 		if ( WPSEO_Options::get( 'noindex-author-noposts-wpseo', true ) ) {
 			unset( $defaults['who'], $defaults['capability'] ); // Otherwise it cancels out next argument.
-			$author_archive                  = new Author_Archive_Helper();
-			$defaults['has_published_posts'] = $author_archive->get_author_archive_post_types();
+			$defaults['has_published_posts'] = YoastSEO()->helpers->author_archive->get_author_archive_post_types();
```

Comment:
**Note for the reviewer**: I refactored this line because the `Author_Archive_Helper` now has a dependency (The `Options_Helper`). Retrieving the helper from the dependency injection container makes sure that it has all the correct dependencies. Even if these dependencies change in the future.

This class already makes use of the helpers surface (`YoastSEO()->helpers`), on line 77. I also checked whether the author sitemap still works as expected (I will add specific tests steps under the impact assessment).